quick_select/QuickSelect.java

```diff
@@ -0,0 +1,63 @@
+public class QuickSelect {
+    /*
+    * partition function
+    * array     : array on which partitioning has to be done
+    * left      : left index of the partitioning subarray
+    * right     : right index of the partitioning subarray
+    * pivotIndex : pivot index from which partition has to done
+    * storeIndex : index of the last element of the left subarray
```

Comment:
I am not sure if `storeIndex` needs to be listed here since it is not a parameter. Same for `pivotIndex` in quickSelect method.